resources/js/home.js

```diff
@@ -1,73 +1,93 @@
-import utils from './utils.js';
+import utils from './utils.js'
+import tableFn from "./table.js"
 
-var fromMenuObj
-var indexContentObj
-var paraHeight
+let paraHeight
+let headerObj, contentObj
 
 
-// Loading Contents for Homepage
-const contentRequest = new XMLHttpRequest()
+/*--------------- Home page content loading section ---------------*/
 const para = document.querySelector('#blog-paragraph')
 const imageArea = document.querySelector('.image-area')
 
-contentRequest.onload = function () {
-  if (this.status === 200) {
-    try {
-      indexContentObj = JSON.parse(this.responseText)
-      para.textContent = indexContentObj.Content
-
-      const image = document.createElement('img')
-      image.src = indexContentObj.image
-      image.className += 'content-img';
-      imageArea.appendChild(image)
-
-      paraResize()
-      
-    } catch {
-      console.warn('JSON not parsed')
-    }
-  } else {
-    console.warn('File not found')
-  }
-}
+utils.jsonCaller('get', 'resources/json/homepage.json', function (object) {
+  para.textContent = object.content
+  // let headerObj = object.table
+  headerObj = object.table
+
+  const image = document.createElement('img')
+  image.src = object.image
+  image.className += 'content-img'
+  imageArea.appendChild(image)
+
+  paraResize()
+  loadTable(headerObj)
+})
 
-contentRequest.open('get', 'resources/json/homepage.json')
-contentRequest.send()
 
+4
+/*--------------- Table loading section ---------------*/
+let tableHeader = document.querySelector("thead")
+let tableBody = document.querySelector('tbody')
```

Comment:
tableHeader and tableBody wont be available here, since thead and tbody are being inserted only later. Unwanted code to be removed

Reply:
Removed unused variables